packages/fluent-ui/src/WrapIfAdditionalTemplate/WrapIfAdditionalTemplate.tsx

```diff
@@ -1,11 +1,79 @@
-import { FormContextType, RJSFSchema, StrictRJSFSchema, WrapIfAdditionalTemplateProps } from '@rjsf/utils';
+import { FocusEvent } from 'react';
+import { TextField } from '@fluentui/react';
+import {
+  ADDITIONAL_PROPERTY_FLAG,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  TranslatableString,
+  WrapIfAdditionalTemplateProps,
+} from '@rjsf/utils';
 
 export default function WrapIfAdditionalTemplate<
   T = any,
   S extends StrictRJSFSchema = RJSFSchema,
   F extends FormContextType = any
 >(props: WrapIfAdditionalTemplateProps<T, S, F>) {
-  const { children } = props;
-  // TODO Implement WrapIfAdditionalTemplate features in FluentUI (#2777)
-  return <>{children}</>;
+  const {
+    children,
+    disabled,
+    id,
+    label,
+    onDropPropertyClick,
+    onKeyChange,
+    readonly,
+    registry,
+    required,
+    schema,
+    style,
+    uiSchema,
+  } = props;
+  const { templates, translateString } = registry;
+
+  // TODO: do this better by not returning the form-group class from master.
+  let { classNames = '' } = props;
+  classNames = 'ms-Grid-col ms-sm12 ' + classNames.replace('form-group', '');
+
+  // Button templates are not overridden in the uiSchema
+  const { RemoveButton } = templates.ButtonTemplates;
+  const keyLabel = translateString(TranslatableString.KeyLabel, [label]);
+  const additional = ADDITIONAL_PROPERTY_FLAG in schema;
+
+  if (!additional) {
+    return (
+      <div id={id} className={classNames} style={{ ...style, marginBottom: 15 }}>
+        {children}
+      </div>
+    );
+  }
+
+  const handleBlur = ({ target }: FocusEvent<HTMLInputElement>) => onKeyChange(target.value);
+
+  return (
+    <div className={classNames} style={{ ...style, marginBottom: 15 }} dir='ltr'>
```

Comment:
I believe you also want the id here too?
```suggestion
    <div id={id} className={classNames} style={{ ...style, marginBottom: 15 }} dir='ltr'>
```

Reply:
Looking at the other themes again, it doesn't look like the `WrapIfAdditionalTemplate`s in mui or antd us the `id` this way. They only use it on the text field (`id={`${id}-key`}`).

I agree that I should have `id={id}` in both `return`s or neither `return`. Preference?

Reply:
Neither return actually. I was looking in the snapshots and there are a ton of duplicate IDs because of this, which is bad HTML. Another good catch!

Reply:
Sounds good. Thanks!